packages/invoices/src/InboundShipment/DetailView/Toolbar.tsx

```diff
@@ -34,17 +41,32 @@ export const Toolbar: FC<ToolbarProps> = ({ draft }) => {
   const t = useTranslation(['replenishment', 'common']);
   const { success, info } = useNotification();
 
-  const { selectedRows } = useTableStore(state => ({
-    selectedRows: (
-      Object.keys(state.rowState)
-        .filter(id => state.rowState[id]?.isSelected)
-        .map(selectedId => data.find(({ id }) => selectedId === id))
-        .filter(Boolean) as InboundShipmentItem[]
-    )
-      .map(({ batches }) => Object.values(batches))
-      .flat()
-      .map(({ id }) => id),
-  }));
+  const { selectedRows } = useTableStore(state => {
+    const { isGrouped } = state;
+
+    if (isGrouped) {
+      return {
+        selectedRows: (
+          Object.keys(state.rowState)
+            .filter(id => state.rowState[id]?.isSelected)
+            .map(selectedId => data.find(({ id }) => selectedId === id))
+            .filter(Boolean) as InboundShipmentItem[]
+        )
+          .map(({ lines }) => lines)
+          .flat()
+          .map(({ id }) => id),
+      };
+    } else {
+      return {
+        selectedRows: (
+          Object.keys(state.rowState)
+            .filter(id => state.rowState[id]?.isSelected)
+            .map(selectedId => lines.find(({ id }) => selectedId === id))
+            .filter(Boolean) as InvoiceLine[]
+        ).map(({ id }) => id),
+      };
+    }
```

Comment:
here, switching between if grouped then the IDs are invoice lines, otherwise inbound items